database-factory/notification.go

```diff
@@ -0,0 +1,126 @@
+package dbfactory
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/mattermost/mattermost-cloud-database-factory/model"
+	"github.com/pkg/errors"
+)
+
+// MMField is used for Mattermost attachement creation
+type MMField struct {
+	Title string `json:"title"`
+	Value string `json:"value"`
+	Short bool   `json:"short"`
+}
+
+// MMAttachment is used to create a Mattermost attachment
+type MMAttachment struct {
+	Fallback   *string    `json:"fallback"`
+	Color      string     `json:"color"`
+	PreText    *string    `json:"pretext"`
+	AuthorName *string    `json:"author_name"`
+	AuthorLink *string    `json:"author_link"`
+	AuthorIcon *string    `json:"author_icon"`
+	Title      *string    `json:"title"`
+	TitleLink  *string    `json:"title_link"`
+	Text       *string    `json:"text"`
+	ImageURL   *string    `json:"image_url"`
+	Fields     []*MMField `json:"fields"`
+}
+
+// MMSlashResponse is used to create the payload for the Mattermost notification
+type MMSlashResponse struct {
+	ResponseType string         `json:"response_type,omitempty"`
+	Username     string         `json:"username,omitempty"`
+	IconURL      string         `json:"icon_url,omitempty"`
+	Channel      string         `json:"channel,omitempty"`
+	Text         string         `json:"text,omitempty"`
+	GotoLocation string         `json:"goto_location,omitempty"`
+	Attachments  []MMAttachment `json:"attachments,omitempty"`
+}
```

Comment:
Instead of manually re-creating these structs you can import the `mattermost-server/v5/model` package

Reply:
Removed structures and reusing mattermost server model.